panic: close of closed channel

```
== found old session: 0xc000014088 &{{0xc000106400}} &{{0xc000106300}} 0 2560 1
recv-sess: # 0 1
connB old: &{{0xc000106300}}

-- close old connB: &{{0xc000106300}}
-- readAWriteB: ctxB close: &{{0xc000106300}} 0 false
-recv new B: 512 [0 0 49 49 54 46 49 57 54 46 56 49 46 49 51 55 58 50 48 48 52 0 0 0 0 0 0 0 0 0 0 0 19 115 97 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 97 100 109 105 110 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 5 55 57 51 52 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 4 3 1 6 10 9 1 0 0 0 0 2 0 0 0 0 0 112 121 109 115 115 113 108 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 7 49 49 54 46 49 57 54 46 56 49 46 49 51 55 58 50 48 48 52 0 0 0 0 0 0 0 0 0 0 0 19 97 100 109 105 110 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 5 4 2 0 0 68 66 45 76 105 98 114 97 114 121 10 0 0 0 0 0 13 17 117 115 95 101 110 103 108 105 115 104 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2 1 0 76 0 0] 12593 0
== found old session: 0xc000014098 &{{0xc000106400}} &{{0xc000106300}} 0 2560 1
recv-sess: # 0 1
connB old: &{{0xc000106300}}

-- close old connB: &{{0xc000106300}}
panic: close of closed channel

goroutine 281 [running]:
main.connB_init({0x52cd30?, 0xc000014058}, 0xc000120120, 0x4?)
	/github/workspace/server.go:98 +0x1ad
main.connB_wait({0x52cd30, 0xc000014058})
	/github/workspace/server.go:137 +0x30b
created by main.main
	/github/workspace/server.go:157 +0x11f

```

--- a/server.go
+++ b/server.go
@@ -150,6 +150,7 @@ func main() {
 		Println("version:", build_version)
 		return
 	}
+	Println("== server\033[32;1m", build_version, "\033[0mstart..")
 
 	sessMap = make(map[uint32]*Session)
 

--- a/server.go
+++ b/server.go
@@ -14,6 +14,7 @@ import (
 	"flag"
 	"fmt"
 	"net"
+	"sync"
 	"time"
 
 	. "pkt"
@@ -36,6 +37,7 @@ type Session struct {
 	connB net.Conn
 	readA chan []byte
 	ctxA  chan int
+	lock  sync.Mutex //lock for sess re-use
 
 	//ack for pkt
 	timer     *time.Timer
@@ -94,6 +96,9 @@ func connB_init(connB net.Conn, sess *Session, need_mid uint16) { // struct must
 	fmt.Println("connB old:", sess.connB)
 
 	// close current connection if it exists
+	sess.lock.Lock()
+	defer sess.lock.Unlock()
+
 	if sess.connB != nil {
 		fmt.Println("\n-- close old connB:", sess.connB)
 		sess.connB.Close() //stop old readBWriteA

